### PR TITLE
Various docs/developer experience fixes

### LINKED ISSUE #45: Clarify whether path matches cross-origin or just same origin
There are some places in the docs that state that `path` will only match same-origin requests, but @wibblymat didn't think that was the case after looking at the `sw-toolbox` source. I need to double-check and most likely update the docs to reflect that cross-origin requests are matched as well.

### LINKED ISSUE #46: Special characters in origin="regex" only need to be single-escaped
The [docs state](https://github.com/PolymerElements/platinum-sw/blob/faf7d8863cf30e8876264ff68c121f32783c903a/platinum-sw-fetch.html#L78) that they need to be double-escaped, but that's not the case.

### LINKED ISSUE #47: Ensure that developers know when they're using an unsupported set of parameters
E.g. defining a `<platinum-sw-fetch>` without a `handler` attribute is meaningless, but there currently isn't any error communicated back to the developer. It's effectively a silent failure.

There is [precedent elsewhere](https://github.com/PolymerElements/platinum-sw/blob/faf7d8863cf30e8876264ff68c121f32783c903a/platinum-sw-register.html#L328) in the project to use `console.error()` to let developers know about this sort of thing, but there's a fine line between alerting developers and spamming the console. It's also not clear how many developers know to check the console.

### LINKED ISSUE #48: Docs: clarify that default cache strategy is applied to local and remote resources
The docs say:

"This is enough to have all of the (local) resources your site uses cached at runtime."

My assumption was that I could't use this for remote resources. From @jeffposnick's research:

When using `<platinum-sw-cache default-cache-strategy="...">`, that handler will be applied to all requests, both same or cross-origin.

### PR DESCRIPTION
**Pull request by jeffposnick**
R: @ebidel @addyosmani @wibblymat 

Much closing!

Closes #45, closes #46, closes #47, closes #48

### CODE REVIEW

File: platinum-sw-fetch.html [hunk 1]
```diff
@@ -122,6 +125,9 @@
         };
         if (this.path && this.handler) {
           params.route = [this.path, this.handler, this.origin];
+        } else {
+          console.log('The following platinum-sw-fetch element will not have any effect. ' +
```
Comment (ebidel) [inline platinum-sw-fetch.html]: console.warn?

File: platinum-sw-import-script.html [hunk 2]
```diff
@@ -49,6 +49,9 @@
         var params = {};
         if (this.href) {
           params.importscript = this.href;
+        } else {
+          console.log('The following platinum-sw-import-script element will not have any effect. ' +
```
Comment (ebidel) [inline platinum-sw-import-script.html]: warn?

**Comment by ebidel** [comment]
LGTM. Just a suggestion. Merge when ready.

**Comment by wibblymat** [comment]
:+1: :shipit:

**Comment by addyosmani** [comment]
Late :+1:

**Pull request merged**